homeassistant/components/media_player/onkyo.py

```diff
@@ -42,6 +43,7 @@
     vol.Optional(CONF_SOURCES, default=DEFAULT_SOURCES):
         {cv.string: cv.string},
     vol.Optional(CONF_ZONE2, default=False): cv.boolean,
+    vol.Optional(CONF_ZONE3, default=False): cv.boolean,
```

Comment:
This feels weird, why not just have it be a number to specify zones ? Or even better, can we query onkyo for the number of zones it has?

Reply:
I just copied the prior art, but you're right - why make this a config when we can provision it. After checking the underlying python library there's no direct way to query about the capabilities of the receiver, but I think I can hack it up.